### PR TITLE
PLAT-165: Migrate test_s3.py

### PR DESCRIPTION
**Pull request by ethho**
- Changes from #1115, #1116, #1117 
- Migrate `test_s3.py`

### CODE REVIEW

**Comment by ethho** [comment]
WIP merge conflicts

**Comment by ethho** [comment]
Done resolving conflicts.

File: tests/conftest.py [hunk 1]
```diff
@@ -12,6 +14,7 @@
 import tempfile
 from datajoint import errors
 from datajoint.errors import ADAPTED_TYPE_SWITCH, FILEPATH_FEATURE_SWITCH
+from datajoint.errors import ADAPTED_TYPE_SWITCH, FILEPATH_FEATURE_SWITCH
```
Comment (ethho) [inline tests/conftest.py:17]: Fixed in a downstream PR

File: tests/test_adapted_attributes.py [hunk 2]
```diff
@@ -18,17 +17,9 @@ def adapted_graph_instance():
...
-
-
-@pytest.fixture
-def enable_filepath_feature(monkeypatch):
-    monkeypatch.setenv(FILEPATH_FEATURE_SWITCH, "TRUE")
-    yield
-    monkeypatch.delenv(FILEPATH_FEATURE_SWITCH, raising=True)
+def schema_name_custom_datatype():
+    schema_name = PREFIX + "_test_custom_datatype"
+    return schema_name
```
Comment (A-Baji) [inline tests/test_adapted_attributes.py:22]: replaced by global constant

File: tests/conftest.py [hunk 3]
```diff
@@ -186,27 +288,76 @@ def schema_adv(connection_test):
...
+    schema(schema_external.SimpleRemote)
+    schema(schema_external.Seed)
+    schema(schema_external.Dimension)
+    schema(schema_external.Image)
+    schema(schema_external.Attach)
+
+    # dj.errors._switch_filepath_types(True)
+    schema(schema_external.Filepath)
+    schema(schema_external.FilepathS3)
+    # dj.errors._switch_filepath_types(False)
```
Comment (A-Baji) [inline tests/conftest.py:310]:
```suggestion
    schema(schema_external.Filepath)
    schema(schema_external.FilepathS3)
```
remove commented code to dodge dmitri aggro

**Review by A-Baji — CHANGES_REQUESTED** [review]

**Pull request merged**